src/main.ml

```diff
@@ -125,6 +125,17 @@ let main port certfile ciphersuites =
          Lwt_unix.bind sock sockaddr;
          Lwt_unix.listen sock 5;
          Lwt_log.notice "Listening for incoming connections" >>= fun () ->
+
+         let conn_count = ref 0 in
+         let conn_m = Lwt_mutex.create () in
+         let inc_conn ?(i=1) () = Lwt_mutex.with_lock conn_m (fun () ->
+           conn_count := !conn_count + i;
+           if !conn_count > Consts.connection_limit && i > 0
```

Comment:
Personally I would find it clearer if we created a separate function for `dec_conn`, instead of reusing the same code. Actually, I think the best would be to create a new wrapper `with_limited_connections`, that way all the related code would be kept together. But this is subjective though.